src/Microsoft.ML.Data/DataLoadSave/Text/TextLoader.cs

```diff
@@ -201,15 +201,21 @@ public sealed class Range
             public Range() { }
 
             public Range(int index)
-                : this(index, index) { }
+            {
+                Contracts.CheckParam(index >= 0, nameof(index), "Must be non-negative");
+                Min = index;
+                Max = index;
+            }
 
-            public Range(int min, int max)
+            public Range(int min, int? max)
             {
-                Contracts.CheckParam(min >= 0, nameof(min), "min must be non-negative.");
-                Contracts.CheckParam(max >= min, nameof(max), "max must be greater than or equal to min.");
+                Contracts.CheckParam(min >= 0, nameof(min), "Must be non-negative");
+                Contracts.CheckParam(!(max < min), nameof(max), "If specified, must be greater than or equal to " + nameof(min));
```

Comment:
I think this would be easier to read/understand if it was written like `CheckParam(max == null || max.Value >= min)`

Reply:
Honestly I'm less enthusiastic about this suggestion than some of your other suggestions. Who exactly is the suggestion meant to serve? That set of people that don't realize that `int?` and `int` are different types?

---
In reply to: [214523116](https://github.com/dotnet/machinelearning/pull/778#discussion_r214523116) [](ancestors = 214523116)

Reply:
> Whom exactly is the suggestion meant to serve? 

People who are reading the code.  IMO this uses a "trick" that `null` is not less than a value, which isn't obvious.

It also aligns the code with the English wording: `If specified, must be greater than or equal to ` directly maps to `max == null || max.Value >= min`

---
In reply to: [214570724](https://github.com/dotnet/machinelearning/pull/778#discussion_r214570724) [](ancestors = 214570724,214523116)